packages/ra-ui-materialui/src/button/SelectAllButton.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useCallback } from 'react';
+import {
+    useListContext,
+    type RaRecord,
+    type UseGetListOptions,
+    type UseReferenceArrayFieldControllerParams,
+    type UseReferenceManyFieldControllerParams,
+} from 'ra-core';
+
+import { Button, ButtonProps } from './Button';
+
+/**
+ * Select all items from a Datagrid.
+ *
+ * @typedef {Object} Props the props you can use
+ * @prop {string} label Button label. Defaults to 'ra.action.select_all', translated.
+ * @prop {string} limit Maximum number of items to select. Defaults to 250.
+ * @prop {function} queryOptions Object of options passed to react-query.
+ *
+ * @param {Props} props
+ *
+ * @example
+ *
+ * const MySelectAllButton = () => <SelectAllButton limit={100} label="Select all books" />;
+ */
+
+const PREFIX = 'RaSelectAllButton';
+
+export const SelectAllButton = (props: SelectAllButtonProps) => {
+    const {
+        label = 'ra.action.select_all',
```

Comment:
I realize we're using the same key for the "Select all" *checkbox* (to select the current page) and the "Select all" *button* (to select all results). I think it's a mistake. Even if the translation may be the same in some languages, in general the labels should be different.